Source/Scene/Model.js

```diff
@@ -1454,6 +1454,191 @@ define([
         return getStringFromTypedArray(loadResources.getBuffer(bufferView));
     }
 
+    function shallowClone(obj) {
+        var newObj = {};
+        for (var key in obj) {
+            if (obj.hasOwnProperty(key)) {
+                newObj[key] = obj[key];
+            }
+        }
+        return newObj;
+    }
+
+    function replaceAllButFirstInString(string, find, replace) {
+        return string.replace(new RegExp(find, 'g'), function(match, offset, all) {
+            return all.indexOf(find) === offset ? match : replace;
+        });
+    }
+
+    function scaleFromMatrix5Array(matrix) {
+        return [matrix[0], matrix[1], matrix[2], matrix[3],
+                matrix[5], matrix[6], matrix[7], matrix[8],
+                matrix[10], matrix[11], matrix[12], matrix[13],
+                matrix[15], matrix[16], matrix[17], matrix[18]];
+    }
+
+    function translateFromMatrix5Array(matrix) {
+        return [matrix[20], matrix[21], matrix[22], matrix[23]];
+    }
+
+    function stringifyArray(array) {
+        return array.map(function(val) {
+            return val.toFixed(20);
+        });
+    }
+    
+    function modifyShaderForQuantizedAttributes(shader, programName, model, context) {
```

Comment:
This function would be cleaner if it was broken into a few functions, perhaps delegate to a new function per primitive in the second nested loop, and then in that function also delegate to a new function to compute `attributeVarName`.

See "Statements in a function should be at a similar level of abstraction..." in the [Coding Guide](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/Documentation/Contributors/CodingGuide#functions).